api/src/main/kotlin/se/ansman/kotshi/NullSafeJsonAdapter.kt

```diff
@@ -0,0 +1,7 @@
+package se.ansman.kotshi
+
+import com.squareup.moshi.JsonAdapter
+
+abstract class NullSafeJsonAdapter<T: Any> protected constructor(private val toString: String) : JsonAdapter<T>() {
```

Comment:
I would call this a NamedJsonAdapter since nothing about it ensures null safety.

Reply:
Yeah, good idea. I named it `NamedJsonAdapter` and added a comment about it not being public.